src/logging.cpp

```diff
@@ -51,8 +84,43 @@ bool BCLog::Logger::StartLogging()
         assert(!m_file_path.empty());
         m_fileout = fsbridge::fopen(m_file_path, "a");
         if (!m_fileout) {
-            return false;
+            return strprintf("Could not open debug log file %s", m_file_path.string());
+        }
+
+        // Special files (e.g. device nodes) may not have a size.
+        try {
+            m_file_size = fs::file_size(m_file_path);
+        } catch (const fs::filesystem_error&) {
+            // We can't do log rotation if it's not a regular file.
+            if (m_rotate_requested) {
+                return strprintf(
+                    "The debug log file %s does not appear to be a regular file; "
+                    "this prevents log rotation. Please disable log rotation by "
+                    "specifying -debuglogrotatekeep=0 or specify a regular file "
+                    "as the debug log file using -debuglogfile.",
+                    m_file_path.string());
+            }
+            m_rotate_keep = 0;
+        }
+        // It gets complicated to add a timestamp to an arbitrary file name,
+        // so require "debug.log".
+        if (m_file_path.filename() != "debug.log") {
+            if (m_rotate_requested) {
+                return strprintf(
+                    "The debug log file %s is not named \"debug.log\"; "
+                    "this prevents log rotation. Please disable log rotation by "
+                    "specifying -debuglogrotatekeep=0 or specify a debug log file "
+                    "path that ends in \"/debug.log\" using -debuglogfile.",
+                    m_file_path.string());
+            }
+            m_rotate_keep = 0;
+        }
+        if (m_file_limit == 0) {
+            return strprintf(
+                "The debug log file %s size limit in MB must be greater than zero.",
+                m_file_path.string());
```

Comment:
i also got a silent failure trying to specify a logsize value < 1. i ran with `-debugloglimit=0.5`, got no errors, but the node failed to start. i also ran with `-debugloglimit=1.5` without issue so i dont think its the decimal

Reply:
That's because it's parsing the argument as an integer, and when it gets to the decimal point, it stops parsing, interpreting the argument as zero. Specifying fractions of a megabyte is probably unnecessary, but maybe it should print a good error message. I'll see what I can do.

Reply:
i went back and confirmed, even with 1.5 it is still rotating in 1mb chunks. might be worth adding something to the option help that debugloglimit is in MB and must be specified as an integer